Make MULTICAST_IP externally configurable using compiler switches

I had to change `MULTICAST_IP` when using Tasmota's KNX support because a router frequently used in Germany (AVM's Fritz!Box) does not correctly implement multicast support. This, however, seems to be a [frequent issue with many routers](https://github.com/knxd/knxd/issues/16). 

As these faulty implementations forward Apple's Bonjour traffic, an option is to set the default multicast address to `224, 0, 0, 251`. Using Tasmota, this requires to modify esp-knx-ip in the `lib/` folder - which should be avoided.

Therefore, it could make sense to change `esp-knx-ip-0.5.2/esp-knx-ip.h` as follows:

Current version:
```
#define MULTICAST_IP              IPAddress(224, 0, 23, 12) // [Default IPAddress(224, 0, 23, 12)]
```

Updated version:
```
#ifndef
#define MULTICAST_IP              IPAddress(224, 0, 23, 12) // [Default IPAddress(224, 0, 23, 12)]
#endif
```

This would allow setting the MULTICAST_IP using compiler switches (e.g., `-DMULTICAST_IP=IPAddress(224, 0, 0, 251)`).

With this setting, I'm able to get a [working setup](https://github.com/knxd/knxd/issues/16#issuecomment-563967604) with KNX/IP to KNX/TP forwarding. So, I'm able to switch a MagicHome LED controller running Tasmota with a KNX/TP switch.

Yes. To verify this setup, I've changed `lib/esp-knx-ip-0.5.2/esp-knx-ip.h` as follows: 
```
#ifndef MULTICAST_IP
#define MULTICAST_IP IPAddress(224, 0, 0, 251)  // [Default IPAddress(224, 0, 23, 12)]
#else
#warning NOT USING DEFAULT SETTING, MULTICAST_IP has been defined before
#endif
```
Afterwards, I've modified my `platformio_override.ini` of my Tasmota build and modified `build_flags` as follows: 

`build_flags = ${core_active.build_flags}  '-DMULTICAST_IP=IPAddress(224, 0, 0, 251)'`.

Compilation output (excerpt):
```
In file included from tasmota/tasmota_post.h:37:0,
                 from /Users/dennis/Documents/dev/Arduino/tasmota/Sonoff-Tasmota-MagicHome/tasmota/tasmota.ino:36:
lib/esp-knx-ip-0.5.2/esp-knx-ip.h:39:2: warning: #warning NOT USING DEFAULT SETTING, MULTICAST_IP has been defined before [-Wcpp]
 #warning NOT USING DEFAULT SETTING, MULTICAST_IP has been defined before
  ^

In file included from lib/esp-knx-ip-0.5.2/esp-knx-ip-config.cpp:7:0:
lib/esp-knx-ip-0.5.2/esp-knx-ip.h:39:2: warning: #warning NOT USING DEFAULT SETTING, MULTICAST_IP has been defined before [-Wcpp]
 #warning NOT USING DEFAULT SETTING, MULTICAST_IP has been defined before
  ^

Compiling .pioenvs/tasmota-knx/lib56f/esp-knx-ip-0.5.2/esp-knx-ip-send.cpp.o
In file included from lib/esp-knx-ip-0.5.2/esp-knx-ip-conversion.cpp:7:0:
lib/esp-knx-ip-0.5.2/esp-knx-ip.h:39:2: warning: #warning NOT USING DEFAULT SETTING, MULTICAST_IP has been defined before [-Wcpp]
 #warning NOT USING DEFAULT SETTING, MULTICAST_IP has been defined before
  ^
Compiling .pioenvs/tasmota-knx/lib56f/esp-knx-ip-0.5.2/esp-knx-ip-webserver.cpp.o
In file included from lib/esp-knx-ip-0.5.2/esp-knx-ip-send.cpp:7:0:
lib/esp-knx-ip-0.5.2/esp-knx-ip.h:39:2: warning: #warning NOT USING DEFAULT SETTING, MULTICAST_IP has been defined before [-Wcpp]
 #warning NOT USING DEFAULT SETTING, MULTICAST_IP has been defined before
  ^
Compiling .pioenvs/tasmota-knx/lib56f/esp-knx-ip-0.5.2/esp-knx-ip.cpp.o
In file included from lib/esp-knx-ip-0.5.2/esp-knx-ip-webserver.cpp:7:0:
lib/esp-knx-ip-0.5.2/esp-knx-ip.h:39:2: warning: #warning NOT USING DEFAULT SETTING, MULTICAST_IP has been defined before [-Wcpp]
 #warning NOT USING DEFAULT SETTING, MULTICAST_IP has been defined before
  ^
In file included from lib/esp-knx-ip-0.5.2/esp-knx-ip.cpp:7:0:
lib/esp-knx-ip-0.5.2/esp-knx-ip.h:39:2: warning: #warning NOT USING DEFAULT SETTING, MULTICAST_IP has been defined before [-Wcpp]
 #warning NOT USING DEFAULT SETTING, MULTICAST_IP has been defined before
```

--- a/lib/esp-knx-ip-0.5.2/esp-knx-ip.h
+++ b/lib/esp-knx-ip-0.5.2/esp-knx-ip.h
@@ -32,8 +32,18 @@
 #define DISABLE_RESTORE_BUTTON    1 // [Default 0] Set to 1 to disable the "restore defaults" button in the web ui.
 
 // These values normally don't need adjustment
-#define MULTICAST_PORT            3671 // [Default 3671]
+#ifndef MULTICAST_IP
 #define MULTICAST_IP              IPAddress(224, 0, 23, 12) // [Default IPAddress(224, 0, 23, 12)]
+#else
+#warning USING CUSTOM MULTICAST_IP
+#endif
+
+#ifndef MULTICAST_PORT
+#define MULTICAST_PORT            3671 // [Default 3671]
+#else
+#warning USING CUSTOM MULTICAST_PORT
+#endif
+
 #define SEND_CHECKSUM             0
 
 // Uncomment to enable printing out debug messages.

